### DISCUSSION TITLE
Is startDate for next occurrence missing?

### DISCUSSION THREAD

**Discussion by TomDeSmet** [solspace/craft-calendar#113] [Questions & Answers] [answered]
Related to #42 I still have an issue with this functionality, namely that we do not get the start date for the next occurrence of a repeating event. I would assume that `startdate` gives us the start date of the next occurrence and `initialStartDate` gives us the start date of the event itself. It seems to me these are always the same. Is this a bug?

**Comment by kjmartens** [comment, marked as answer]
Sorry for the delay @TomDeSmet.

I'm not sure what your full context is. However, in a general context, the issue with events displaying a start date specific to their occurrence date is that Calendar inflates results based on a repeat rule code, so there are no unique event entries in the database for each recurrence. So when accessing an event based on its ID or slug, Calendar has no way of knowing which occurrence you're referring to. There is a way around it though, by including date segments in the URL:
https://docs.solspace.com/craft/calendar/v3/introduction.html#getting-around-recurrence-date-limitations-on-event-detail-page

I'm not sure if this is helpful to your question. I'd need to know more about your setup and flow. 🙂
